### PR TITLE
Refactoring business logic into classes in $lib/business

### PR DESCRIPTION
**Pull request by Erim32**
Refactoring business logic into classes to simplify function reuse and maintenance.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `UserProfileManager` class for streamlined user profile data management.
	- Enhanced user profile retrieval process using the new `UserProfileManager` in various application areas.

- **Bug Fixes**
	- Improved logic in the `updateEmail` action to accurately determine if a profile is new based on its existence and update status. 

- **Refactor**
	- Streamlined the `load` function and `updateEmail` action by encapsulating user profile fetching within the `UserProfileManager`, improving code maintainability and organization.

### CODE REVIEW

**Comment by scosman** [comment]
Not sure how I feel about this one. Fetching profiles from supabase is a 1 liner and nicely returns typed data and error data. Now it's 2 lines inline, and a wrapper class with state.

Supabase API is pretty good. If you have custom needs (only specific fields, join with another table), you can do that easily. Anyone adopting supabase will need to learn the query syntax, and I don't want to encourage wrapping all calls to it, or hiding it.

Yes, the same code appears twice but: https://go-proverbs.github.io `A little copying is better than a little dependency`

File: src/lib/business/UserProfileManager.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
+ * User Profile Manager
+ * This class is responsible for managing the user profile data
+ */
+export class UserProfileManager {
+  private supabase: any
+  private session: any
+
+  constructor(supabase: any, session: any) {
+    this.supabase = supabase
```
Comment (scosman) [inline src/lib/business/UserProfileManager.ts:10]: We should keep strong typing for these if we want to take this.

Comment (Erim32) [reply src/lib/business/UserProfileManager.ts:10]: Totally agree, I just didn't find the class to import.

Comment (scosman) [reply src/lib/business/UserProfileManager.ts:10]: See https://github.com/CriticalMoments/CMSaasStarter/pull/108#issuecomment-2254491650 before working on it though, should come up with the strategy!

**Comment by Erim32** [comment]
> Not sure how I feel about this one. Fetching profiles from supabase is a 1 liner and nicely returns typed data and error data. Now it's 2 lines inline, and a wrapper class with state.
> 
> Supabase API is pretty good. If you have custom needs (only specific fields, join with another table), you can do that easily. Anyone adopting supabase will need to learn the query syntax, and I don't want to encourage wrapping all calls to it, or hiding it.
> 
> Yes, the same code appears twice but: https://go-proverbs.github.io `A little copying is better than a little dependency`

Correct si nous ne voyons que la base de code du template, mais si l'utilisation du template nécessite plusieurs pages/routes cela signifie de dupliquer le code.
L'objectif est d'améliorer la maintenabilité et rajouter une abstraction si par exemple certains utilisateurs veulent un jour remplacer toute les références de supabase par une autre solution

**Comment by scosman** [comment]
Supabase *is* our database model abstraction layer. It has the types, it has the select methods, it has the update methods, it has the constraints, etc.

A call to the supabase style of "get profile" is a bit more verbose, but that's because it's a powerful API that can do many things (select many profiles, select all profiles updated in last 24 hours, select only select fields from a profile, etc). If you're going to use supabase, it's good to learn its API. Happening to make a call to get the profile twice isn't repeated code, as much as repeated parameters. It's actually fewer LOC inline (1 vs 2), and much less code overall. 

"for example, some users one day want to replace all supabase references with another solution". Abstracting for a "some users one day" would be a premature abstraction. We'd have to abstract a lot more supabase code to make it possible to swap out the backend. I'm not sure how I feel about making the template that abstract. If we did go that route, it's not something we'd want to do one piece at a time; we'd wait until we had it all done in a branch, and at least 1 real implementation of another backend to justify the additional complexity/abstraction today.

**Comment by scosman** [comment]
@Erim32 I take you you're not working on these 3 PRs anymore? Let me know. If so I'll close them out just to keep things clean.

**Comment by scosman** [comment]
Closing as I haven't heard back from creator: https://github.com/CriticalMoments/CMSaasStarter/pull/108#issuecomment-2318400900

**Pull request closed without merging**